packages/slate/test/models/text/delete/all-text-length/marked-text.js

```diff
@@ -0,0 +1,10 @@
+/** @jsx h */
+
+import h from '../../../../helpers/h'
+
+export const input = <b>Cat is Cute</b>[0]
+export default function(t) {
+  return t.removeText(0, t.text.length)
+}
+
+export const output = <b />[0]
```

Comment:
I'd like to revert this behavior. It should be the same as deleting all the characters above with italic and bold, where the end marks are none.

Reply:
I am considering this case: 

If we have some blocks are marked with bold or other marks, like:
```
<b>
<p>First Line</p>
<p>Second Line</p>
<p>Third Line</p>
</b>
```
If someone delete the all content in the second line, then insert, under the revert, the user will get
```
<p><b>First Line</b></p>
<p>New Second Line</p>
<p><b>Third Line</b></p>
```

By keeping the mark, the user will get the second line bold.
```
<p><b>First Line</b></p>
<p><b>New Second Line</b></p>
<p><b>Third Line</b></p>
``` 
Some aged users tends to select all document, mark with times-new-roman font, and then starts editing.  They will think that, if a range is marked, then any insert in that range is always marked.  So I implement this feature for empty text.


The only potential bug of the empty text with mark is about merging text node and insert text; I have made the special treatment that, if the current leaf text is empty, then the mark will have no influence if the merged node or the insert text is not empty.

Reply:
Good argument, it looks like Google Docs follows your logic here too. I'm convinced!